### PR TITLE
Create a random new core for every integration test

### PR DESCRIPTION
**Pull request by thomascorthals**
As proposed by @mkalkbrenner in #750, I modified `AbstractCoreTest` to create a new uniquely named core for each integration test and unload it afterwards.

The setUp and tearDown always use the default Curl adapter. I didn't look into switching adapters at this point because the ping test in `setUp()` always uses Curl too.

Fresh cores don't have a `lastModified` time yet, from my empirical findings. I couldn't find official Solr documentation on when exactly this time does get set, so I'm allowing either `null` or a `\DateTime` for that assertion. You can verify the `isInstanceOf` part by setting the core to `'techproducts'` for that test.

This PR doesn't solve the testing issue we ran into in #750 because the new cores share the same configset. We still end up with artifacts if the previous core's config wasn't cleaned up properly by Solr.

### CODE REVIEW

File: tests/Integration/AbstractCoreTest.php [hunk 1]
```diff
@@ -19,22 +19,64 @@ abstract class AbstractCoreTest extends AbstractTechproductsTest
...
+
+            // create core with unique name using the techproducts configset
+            $createAction = $coreAdminQuery->createCreate();
+            $createAction->setCore(self::$core);
+            $createAction->setConfigSet('sample_techproducts_configs');
+            $coreAdminQuery->setAction($createAction);
+            $response = $client->coreAdmin($coreAdminQuery);
+            static::assertTrue($response->getWasSuccessful());
+
+            // merge techproducts index into our new core
```
[the remark this answers is not in the document]
Comment (thomascorthals) [reply tests/Integration/AbstractCoreTest.php]: That was my first idea as well. Import from where though? I don't think that it's a good idea to try and read the XMLs with example data in the Solr directory tree from within the unit tests. Should we include them in `tests/Integration/Fixtures`?

[the remark this answers is not in the document]
Comment (mkalkbrenner) [reply tests/Integration/AbstractCoreTest.php]: Including them seems to be the right solution.

[the remark this answers is not in the document]
Comment (thomascorthals) [reply tests/Integration/AbstractCoreTest.php]: The XMLs contain valid `<add>` commands. Easiest way to import is to add a command type `COMMAND_RAW` to `QueryType\Update\Query`. Or is it "purer" Solarium to parse the XML and rebuild it through `COMMAND_ADD`?

[the remark this answers is not in the document]
Comment (mkalkbrenner) [reply tests/Integration/AbstractCoreTest.php]: go for it :-)

**Comment by mkalkbrenner** [comment]
Since we can't work together on this PR I created #780

**Pull request closed without merging**